src/Common/src/TypeSystem/Common/FieldDesc.cs

```diff
@@ -69,6 +69,13 @@ public virtual string Name
             get;
         }
 
+        /// <summary>
+        /// Fields in AOT can be preinitialized with special RVA fields (indicated by InitDataBlobAttribute)
+        /// whose RVA points to the contents of the value (such as array).
+        /// Returns the corresponding field that contains preinitialized data.
+        /// </summary>
+        public virtual FieldDesc PreInitDataField => null;
```

Comment:
Great point. I'm also a bit unsure about whether I should put the preinitialized information (and learning things as I go) so putting on FieldDesc seems to be the most natural choice at that moment :) But you are right and I agree - moving this into its own helper class seems to be a better choice, and abstracting out the details of where the preinitialized data are is also very nice . I was mostly worry about caching the result but it looks like a natural place to avoid the duplicate calculation is in EETypeNode. I'll make the change